src/main/java/com/allocadia/carbonite/CarbonitePersist.java

```diff
@@ -0,0 +1,54 @@
+package com.allocadia.carbonite;
+
+import org.springframework.jdbc.core.support.JdbcDaoSupport;
+
+import com.allocadia.carbonite.utils.QueryUtils;
+import com.google.common.base.Joiner;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import lombok.SneakyThrows;
+
+public class CarbonitePersist<T extends PersistedObject> extends JdbcDaoSupport {
+
+    public Collection<T> save(Carbonite carbonite, Collection<T> objects) {
+        ModifiedObjects modifiedObjects = ModifiedObjects.from(objects);
+        Class<?> clazz = objects.iterator().next().getClass();
+        PersistenceInfo<PersistedObject> info = (PersistenceInfo<PersistedObject>) carbonite.getPersistenceInfo(clazz);
+        
+        persistNew(modifiedObjects.getNewObjects(), info);
+        //TODO: persistDirty
+    }
+
+    @SneakyThrows
+    private void persistNew(List<PersistedObject> newObjects, PersistenceInfo<PersistedObject> info) {
+        Map<Field, String> field2Column = info.getField2Column();
+        String baseSql = Joiner.on("\n").join(
+            "INSERT INTO " + info.getTableName(),
+            "(" + String.join(", ", field2Column.values()) + ")",
+            "VALUES",
+            "(" + QueryUtils.createParamList(field2Column.values().size()) + ")");
+        
+        List<Object[]> argList = new ArrayList<>();
+        for (PersistedObject persistedObject : newObjects) {
+            Object[] objectArgs = new Object[field2Column.keySet().size()];
+            
+            Iterator<Field> fieldIterator = field2Column.keySet().iterator();
+            int position = 0;
+            while (fieldIterator.hasNext()) {
+                objectArgs[position++] = fieldIterator.next().get(persistedObject);
+            }
+            
+            argList.add(objectArgs);
+        }
+        
+        super.getJdbcTemplate().batchUpdate(baseSql, argList);
```

Comment:
If we end up doing a fetch-id query after each insert query, maybe we may as well fetch the full row to also retrieve default DB column values?
```sql
INSERT INTO ... (..dirty-fields..) VALUES (...);
SELECT ..fields-not-in-^... FROM ... WHERE ID = LAST_INSERT_ID();
```
Or something like that... Would that be too expensive compared to just `SELECT LAST_INSERT_ID()`?

Would be a lot nicer then forcing those defaults to be in two places (sql + java) or magically detecting those defaults with something such as `DESCRIBE TABLE ...` queries.